dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java

```diff
@@ -1707,23 +1707,6 @@ public void addResolvedExpressionItem( DimensionalItemObject item)
         {
             resolvedExpressionItems.add(item);
         }
-        else
-        {
-            checkForIndicatorCyclicReference(item);
-        }
-    }
-
-    private void checkForIndicatorCyclicReference( final DimensionalItemObject item )
-    {
-        final boolean isNestedIndicator = item instanceof Indicator
-            && (((Indicator) item).getNumerator().contains( NESTED_INDICATOR_PREFIX )
-                || ((Indicator) item).getDenominator().contains( NESTED_INDICATOR_PREFIX ));
-        if ( isNestedIndicator )
-        {
-            throw new CyclicReferenceException(
-                String.format( "Item of type %s with identifier '%s' has a cyclic reference to another item.",
-                    item.getDimensionItemType().name(), item.getUid() ) );
-        }
     }
 
     private void addResolvedExpressionItems( List<DimensionalItemObject> dimensionalItemObjectList )
```

Comment:
I think you can remove this method as well and also `public Builder withResolvedExpressionItems` and associated logic